scandium-core/src/test/java/org/eclipse/californium/scandium/DTLSConnectorTest.java

```diff
@@ -527,7 +530,9 @@ public void testAcceptClientHelloAfterIncompleteHandshake() throws Exception {
 		client = new DTLSConnector(clientConfig, clientConnectionStore);
 
 		// THEN assert that the handshake succeeds and a session is established
-		givenAnEstablishedSession();
+		// sometime the retransmission of the previous incomplete handshake
+		// fails the new handshake. Therefore a retry is used.
+		givenAnEstablishedSessionWithRetry();
```

Comment:
> sometime the retransmission of the previous incomplete handshake fails the new handshake.

I'm curious : how does it fail ?

Reply:
First handshake is in progress. Potential, the last pending flight is retransmitted.
Second handshake sends the 2. client_hello. If now the server_hello from the first is received and processed, the handshake fails. There is a very small window, when the last pending flight is resend and the new client_hello is received on the server side. You may use the repeated test runner to see, that without the retry, the test fails rarely.

Reply:
Ok I see.